src/dodal/devices/oav/oav_detector.py

```diff
@@ -204,12 +204,17 @@ class OAV(AreaDetector):
     def __init__(self, *args, params: OAVConfigParams, **kwargs):
         super().__init__(*args, **kwargs)
         self.parameters = params
+        self.subscription_id = None
 
     def wait_for_connection(self, all_signals=False, timeout=2):
         connected = super().wait_for_connection(all_signals, timeout)
         x = self.snapshot.x_size.get()
         y = self.snapshot.y_size.get()
 
         cb = partial(self.parameters.update_on_zoom, xsize=x, ysize=y)
-        self.zoom_controller.level.subscribe(cb)
+
+        if self.subscription_id:
+            self.zoom_controller.level.unsubscribe_all()
```

Comment:
Should: As discussed, might be better just to unsubscribe the one we want otherwise we're causing side effects?